### DISCUSSION TITLE
Surface source/carrier on delivery option

### DISCUSSION THREAD

**Discussion by gil--** [Shopify/function-examples#464] [Feedback: Delivery Customization API]
Currently we have access to the delivery option (shipping rate) source when running shipping discounts via the Script editor: 

![image](https://github.com/Shopify/function-examples/assets/3484527/fe49a730-b0e8-4046-92bc-48a9c3c88522)

This exists on UI Extensions but not Functions.
![image](https://github.com/Shopify/function-examples/assets/3484527/0e2f86c2-866e-4307-988b-9a7ef9e20521)

API: https://shopify.dev/docs/api/functions/reference/shipping-discounts/graphql/common-objects/cartdeliveryoption

**Comment by jasnlink** [comment]
This is needed and I hope to be able to have access to this field.

Right now, there is no way to differentiate between the different delivery options and associate them back to the delivery services set up in the shop. Especially if they have the same titles.

The handle is encoded and can only be used to differentiate the options.
The code can be used to differentiate the options but can't be used to associate back to the service that is set up in the shop because we can't access it from the DeliveryParticipant or DeliveryRateDefinition.
